src/ansys/simai/core/data/global_coefficients_requests.py

```diff
@@ -0,0 +1,209 @@
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import logging
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any, TypeVar
+
+from ansys.simai.core.data.base import (
+    ERROR_STATES,
+    PENDING_STATES,
+    ComputableDataModel,
+    Directory,
+)
+
+if TYPE_CHECKING:
+    import ansys.simai.core.client
+
+EXTRA_CALCULETTE_FIELDS = ["Area", "Normals", "Centroids"]
+
+logger = logging.getLogger(__name__)
+
+
+class GlobalCoefficientRequest(ABC, ComputableDataModel):
+    """Creates the basic request for making requests related to Global Coefficients."""
+
+    def __init__(
+        self,
+        client: "ansys.simai.core.client.SimAIClient",
+        directory: "Directory",
+        fields: dict,
+        project_id: str,
+        gc_formula: str,
+        sample_metadata: dict[str, Any],
+        bc: list[str] = None,
+        surface_variables: list[str] = None,
+        **kwargs,
+    ):
+        super().__init__(client=client, directory=directory, fields=fields)
+
+        self.project_id = project_id
+        self._calculette_payload = self._compose_calculette_payload(
+            gc_formula, sample_metadata, bc, surface_variables
+        )
+
+    @ComputableDataModel._failure_message.getter
+    def _failure_message(self):
+        return self.fields.get("error")
+
+    def _compose_calculette_payload(
+        self,
+        gc_formula: str,
+        sample_metadata: dict[str, Any],
+        bc: list[str] = None,
+        surface_variables: list[str] = None,
+    ):
+        """Composes the payload for a calculette request."""
+
+        surface_vars_list = EXTRA_CALCULETTE_FIELDS
+        if surface_variables:
+            surface_vars_list += surface_variables
+
+        return {
+            "formula": gc_formula,
+            "bc_list": bc if bc else [],
+            "surface_field_list": [
+                fd
+                for fd in sample_metadata.get("surface").get("fields")
+                if fd.get("name") in surface_vars_list
+            ],
+            "volume_field_list": [],
+        }
+
+    @abstractmethod
+    def run(self):
+        """Abstract method to perform the customized required process."""
+
+
+GlobalCoefficientRequestType = TypeVar(
+    "GlobalCoefficientRequestType", bound=GlobalCoefficientRequest
+)
+
+
+class GlobalCoefficientRequestDirectory(Directory[GlobalCoefficientRequestType]):
```

Comment:
`CheckGlobalCoefficientDirectory` and `ComputeGlobalCoefficientDirectory` follow the same pattern and nomenclature of the rest of the code in interfaces that handle server requests. This is to secure coherency and stability. They are directories of SSE events too. I see your point regarding the naming (if I understand correctly), but any refactoring of this type could come to a later stage.

Reply:
> CheckGlobalCoefficientDirectory and ComputeGlobalCoefficientDirectory follow the same pattern and nomenclature of the rest of the code in interfaces that handle server requests. 

We talked about it a bit and I agree that it's fine as is for now but I wouldn't agree with the above statement.
Directory is a class built for server objects with public CRUD operations. Currently our SSE behaviors are tied to it because we never had a need for SSE outside of it. 
But there is in the code base at least an example of a class similar to directory but without the data model: https://github.com/ansys/pysimai/blob/df387bfcab19ad45efbd911384aa0e882eca01ec/src/ansys/simai/core/data/design_of_experiments.py#L34
For a refactoring we could move the SSE behaviors to a separate class and then compose where needed: on what could become the `GlobalCoefficientCollection` (or something else) and on the existing directories. Maybe we'll also need to do the same for the SSE part of `ComputableDataModel`.